eva_cttv_pipeline/evidence_string_generation/clinvar_to_evidence_strings.py

```diff
@@ -310,25 +310,27 @@ def skip_record(clinvar_record, clinvar_record_measure, consequence_type, allele
 
 
 def get_consequence_types(clinvar_record_measure, consequence_type_dict):
-    alt_str = clinvar_record_measure.alt if clinvar_record_measure.alt is not None else "-"
-    coord_id = "{}:{}-{}:1/{}".format(clinvar_record_measure.chr, clinvar_record_measure.start,
-                                      clinvar_record_measure.stop, alt_str)
-
-    consequence_type_dict_id = None
-
-    if clinvar_record_measure.rs_id is not None and \
-                    clinvar_record_measure.rs_id in consequence_type_dict:
-        consequence_type_dict_id = clinvar_record_measure.rs_id
-    elif clinvar_record_measure.nsv_id is not None and \
-                    clinvar_record_measure.nsv_id in consequence_type_dict:
-        consequence_type_dict_id = clinvar_record_measure.nsv_id
-    elif coord_id in consequence_type_dict:
-        consequence_type_dict_id = coord_id
-    elif clinvar_record_measure.clinvar_record.accession in consequence_type_dict:
-        consequence_type_dict_id = clinvar_record_measure.clinvar_record.accession  # todo change this depending upon OT gene mapping file
-
-    return consequence_type_dict[consequence_type_dict_id] \
-        if consequence_type_dict_id is not None else [None]
+    """Return the list of functional consequences for a given ClinVar record measure."""
+
+    # First, we attempt to pair by coordinates (only if they are present in the record measure)
+    if clinvar_record_measure.has_complete_coordinates:
+        # This VCF-flavoured identifier is used to pair ClinVar records with functional consequence predictions.
+        # Example of such an identifier: 14:23423715:G:A
+        coord_id = ':'.join([clinvar_record_measure.chr, str(clinvar_record_measure.vcf_pos),
+                             clinvar_record_measure.vcf_ref, clinvar_record_measure.vcf_alt])
+        if coord_id in consequence_type_dict:
+            return consequence_type_dict[coord_id]
+
+    # If no luck, try to pair by ClinVar record accession. This is not a good option because ClinVar accessions do not
+    # necessarily correspond to single variants. This pairing is only retained for compatibility purposes to handle
+    # trinucleotide repeat expansion variants records, which are added manually using RCV identifiers. This will be
+    # removed as soon as correct handling of NT repeat variants is implemented.
```

Comment:
```suggestion
    # removed as soon as correct handling of TN repeat variants is implemented.
```
?

Reply:
NT here stands for “nucleotide”. (Yes, I too think that this abbreviation is annoying, but it is unfortunately widely used)